v3/src/hooks/use-keyboard-shortcuts.ts

```diff
@@ -0,0 +1,39 @@
+import { useEffect } from "react"
+import { tinykeys } from "tinykeys"
+import { appState } from "../models/app-state"
+
+export const useKeyboardShortcuts = () => {
+  useEffect(function handleKeyboardShortcuts() {
+
+    const isEditable = (el: Element | null) => {
+      if (!el) return false
+      if (el instanceof HTMLInputElement ||
+        el instanceof HTMLTextAreaElement ||
+        (el instanceof HTMLElement && el.isContentEditable)) {
+        return true
+      }
+    }
+
+    const okToUndoRedo = (e: KeyboardEvent, test:boolean) => {
```

Comment:
```suggestion
    const okToUndoRedo = (e: KeyboardEvent, test: boolean) => {
```
Also, this code would benefit from a comment indicating why editability matters.